### PR TITLE
Creating a toolchain package docs

### PR DESCRIPTION
**Pull request by czoido**
Example in https://github.com/conan-io/examples2/pull/138
Closes: https://github.com/conan-io/docs/issues/3208

### CODE REVIEW

File: tutorial/creating_packages/other_types_of_packages/toolchain_packages.rst [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+            self.run(f"{toolchain}-gcc --version")
+            test_file = os.path.join(self.cpp.build.bindirs[0], "test_package")
+            stdout = StringIO()
+            self.run(f"file {test_file}", stdout=stdout)
+            if toolchain == "aarch64-none-linux-gnu":
+                assert "ELF 64-bit" in stdout.getvalue()
+            else:
+                assert "ELF 32-bit" in stdout.getvalue()
+
+This test package ensures that the toolchain is functional building a minimal *hello world*
```
Comment (RubenRBS) [inline tutorial/creating_packages/other_types_of_packages/toolchain_packages.rst:364]:
```suggestion
This test package ensures that the toolchain is functional, building a minimal *hello world*
```

File: tutorial/creating_packages/other_types_of_packages/toolchain_packages.rst [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+    $ cd .. && cd consumer
+    $ conan install . -pr:b=default -pr:h=../profiles/raspberry-64 -pr:h=../profiles/arm-toolchain --build missing
+    $ cd build && cmake .. -DCMAKE_TOOLCHAIN_FILE=Release/generators/conan_toolchain.cmake -DCMAKE_BUILD_TYPE=Release
+    $ cmake --build .
+    $ file compressor 
+    compressor: ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), dynamically
+    linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info,
+    not stripped
+
+We composed the already existing profile with other profile that just has
```
Comment (RubenRBS) [inline tutorial/creating_packages/other_types_of_packages/toolchain_packages.rst:432]:
```suggestion
We composed the already existing profile with another profile called ``arm-toolchain`` that just has
```

**Review by RubenRBS — APPROVED** [review]
Nice!

File: tutorial/creating_packages/other_types_of_packages/toolchain_packages.rst [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+            })
+
+Validating the toolchain package: settings, settings_build and settings_target
+------------------------------------------------------------------------------
+
+As you may recall, the :ref:`validate() method<reference_conanfile_methods_validate>` is
+used to indicate that a package is not compatible with certain configurations. As
+mentioned earlier, we are limiting the usage of this package to a *Linux x86_64* platform
+for cross-compiling to a *Linux ARM* target, supporting both 32-bit and 64-bit
+architectures. Let's check how we incorporate this information into the ``validate()``
```
Comment (memsharded) [inline tutorial/creating_packages/other_types_of_packages/toolchain_packages.rst:135]: I think this section will greatly benefit from a graphviz representation of the build/host/target contexts

File: tutorial/creating_packages/other_types_of_packages/toolchain_packages.rst [hunk 4]
```diff
@@ -0,0 +1,448 @@
+.. _tutorial_other_toolchain_packages:
+
+
+Creating a Conan package for a toolchain
```
[the remark this answers is not in the document]
Comment (czoido) [reply tutorial/creating_packages/other_types_of_packages/toolchain_packages.rst:4]: Yes, I had the same feeling... another idea would be to move it to an example, and also move the part that uses the toolchain to the tutorial that uses the profile linking to the example...

[the remark this answers is not in the document]
Comment (memsharded) [reply tutorial/creating_packages/other_types_of_packages/toolchain_packages.rst:4]: Yes, that would make sense, having the "usage" of the toolchain in the tutorial, and the full explanation of the recipe, the ``setting_target``, etc, in its dedicated section in the examples.

[the remark this answers is not in the document]
Comment (czoido) [reply tutorial/creating_packages/other_types_of_packages/toolchain_packages.rst:4]: great! I'll move it

File: tutorial/creating_packages/other_types_of_packages/toolchain_packages.rst [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+
+    from conan import ConanFile
+    from conan.tools.cmake import CMake, cmake_layout
+
+
+    class TestPackgeConan(ConanFile):
+        settings = "os", "arch", "compiler", "build_type"
+        generators = "CMakeToolchain", "VirtualBuildEnv"
+        test_type = "explicit"
+
```
Comment (czoido) [inline tutorial/creating_packages/other_types_of_packages/toolchain_packages.rst:338]:
```suggestion
    class TestPackageConan(ConanFile):
        settings = "os", "arch", "compiler", "build_type"
        generators = "CMakeToolchain", "VirtualBuildEnv"

```

File: tutorial/creating_packages/other_types_of_packages/toolchain_packages.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (czoido) [reply tutorial/creating_packages/other_types_of_packages/toolchain_packages.rst]: In the end I moved it to an example, but I left the part of using it in the example and linked it in the tutorial to not extend the tutorial too much

**Review by memsharded — APPROVED** [review]
Looks good

**Pull request merged**